src/server.rs

```diff
@@ -0,0 +1,46 @@
+//! The application's web server.
+//! Serves both api and web traffic.
+
+use crate::{
+    api::{self, ApiState, ApiStateInner},
+    models::forum::Forum,
+    resources::DatabaseClient,
+    web::{pages, WebState, WebStateInner},
+};
+use axum::{Extension, Router};
+
+pub async fn run() -> color_eyre::Result<Router> {
+    let database = DatabaseClient::new()?;
+    let database_health_check =
+        health::PeriodicChecker::new(database.clone(), health::Config::default());
+    tokio::spawn(database_health_check.clone().run());
+
+    let web_state = WebState {
+        inner: WebStateInner {
+            forums: vec![Forum {
+                id: "asdf",
+                name: "hello world",
+                description: "this is a forum",
+            }],
+        }
+        .into(),
+    };
+
+    let api_state = ApiState {
+        inner: ApiStateInner {
+            forums: vec![Forum {
+                id: "asdf",
+                name: "hello world",
+                description: "this is a forum",
+            }],
+        }
+        .into(),
+    };
+
+    let router = Router::new()
+        .nest("/", pages::compose(web_state))
+        .nest("/api/v1", api::routes::v1::compose(api_state))
```

Comment:
**nitpick**: I don't particularly like the API route, with versioning and such, set up in `main.rs`. Would seem better to create a subrouter for `/api` that would attach `/v1`, and in the future `/v2`, `/v3`, etc.

Reply:
Great point, I think I prefer this too. I reworked this a bit to mount both /healthz and /v1 down in the api module. waffling a bit on where best to put /healthz, but thought this worked cleanly for now

Reply:
I could lean either way; I only wanted to push back a smidge on the `/v1` mount, as it keeps it modular for if/when we split the API into a separate project vs the plans for the prototype monolith. in that state of mind, makes more sense to keep the health check under the API router, so happy with that.